### PR TITLE
AAE-1068 - fixed process extensions schema

### PR DESCRIPTION
**Pull request by VitoAlbano**

### CODE REVIEW

File: activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/resources/schema/process-extensions-schema.json [hunk 1]
```diff
@@ -229,7 +245,7 @@
           ]
         },
         "value": {
-          "type": ["string", "number", "integer", "boolean", "object", "array"]
```
Comment (pmartinezga) [inline activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/resources/schema/process-extensions-schema.json]: We modified this in this PR: https://github.com/Activiti/activiti-cloud-modeling-service/pull/474

We need this not to be a string to allow values in output mappings

File: activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/test/java/org/activiti/cloud/services/modeling/converter/JsonSchemaFlattenerTest.java [hunk 2]
```diff
@@ -13,102 +13,95 @@
 
 public class JsonSchemaFlattenerTest {
 
-   private JsonSchemaFlattener flattener = new JsonSchemaFlattener();
+    private JsonSchemaFlattener flattener = new JsonSchemaFlattener();
 
-   @Test
-   public void should_flattenProcessExtensionSchema() throws IOException {
+    @Test
+    public void should_flattenProcessExtensionSchema() throws IOException {
```
Comment (zoltanpalfi) [inline activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/test/java/org/activiti/cloud/services/modeling/converter/JsonSchemaFlattenerTest.java:19]: Could you please format the file so the extra spaces are no longer?

File: activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/test/java/org/activiti/cloud/services/modeling/converter/JsonSchemaFlattenerTest.java [hunk 3]
```diff
@@ -13,102 +13,95 @@
...
+        refObject.put("$ref", classPath);
 
-       return object;
-   }
+        JSONArray arrObject = new JSONArray();
+        arrObject.put(refObject);
 
-   private JSONObject getTestJSONObjectWithDefinitions(String classPath) {
-       JSONObject object = new JSONObject();
+        JSONObject obj = new JSONObject();
```
Comment (zoltanpalfi) [inline activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/test/java/org/activiti/cloud/services/modeling/converter/JsonSchemaFlattenerTest.java]: variable never used

File: activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/test/java/org/activiti/cloud/services/modeling/converter/JsonSchemaFlattenerTest.java [hunk 4]
```diff
@@ -13,102 +13,95 @@
...
 
-       return new JSONObject(new JSONTokener(new ClassPathResource(schemaFileName)
-                                          .getInputStream()));
-   }
+        JSONArray arrObject = new JSONArray();
+        arrObject.put(refObject);
 
-   private JSONObject getTestJSONObjectWithoutDefinitions(String classPath) {
-       JSONObject object = new JSONObject();
+        JSONObject obj = new JSONObject();
```
Comment (zoltanpalfi) [inline activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/test/java/org/activiti/cloud/services/modeling/converter/JsonSchemaFlattenerTest.java]: variable never used

**Review by zoltanpalfi — APPROVED** [review]

File: activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/resources/schema/process-extensions-schema.json [hunk 5]
```diff
@@ -6,47 +6,22 @@
...
-          "type": "object",
-          "additionalProperties": {
-            "type": "object",
-            "additionalProperties": {
-              "type": "object",
-              "$ref": "#/definitions/constants"
-            }
-          }
-        }
+      "additionalProperties": false,
```
Comment (ffazzini) [inline activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/resources/schema/process-extensions-schema.json:9]: in the ticket description there is a different type of json , am i missing anything?

Comment (VitoAlbano) [reply activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/resources/schema/process-extensions-schema.json:9]: This is just the first step , this is enhancing the validation that we have for the current extension structure, i have another PR that will upgrade this schema once the FE app is ready.

**Review by ffazzini — APPROVED** [review]

**Pull request merged**